gui.py

```diff
@@ -0,0 +1,418 @@
+    ckpt_folder_entry.insert(tk.END, ckpt_location)
+
+
+def launch_training():
+    model_type = model_type_combobox.get()
+    save_optimizer_state = save_optimizer_state_var.get()
+    resume = resume_var.get()
+    project_name = project_name_entry.get()
+    learning_rate = learning_rate_entry.get()
+    match_text_to_unet = match_text_to_unet_var.get()
+    text_lr = text_lr_entry.get()
+    schedule = schedule_combobox.get()
+    text_lr_scheduler = text_lr_scheduler_combobox.get()
+    lr_warmup_steps = lr_warmup_steps_entry.get()
+    lr_decay_steps = lr_decay_steps_entry.get()
+    text_lr_warmup_steps = text_lr_warmup_steps_entry.get()
+    text_lr_decay_steps = text_lr_decay_steps_entry.get()
+    resolution = resolution_slider.get()
+    batch_size = batch_size_entry.get()
+    gradient_steps = gradient_steps_entry.get()
+    dataset_location = dataset_entry.get()
+    max_epochs = max_epochs_slider.get()
+    sample_steps = sample_steps_entry.get()
+    sample_file = sample_file_combobox.get()
+    clip_grad_norm = clip_grad_norm_var.get()
+    disable_amp = disable_amp_var.get()
+    disable_textenc_training = disable_textenc_training_var.get()
+    disable_xformers = disable_xformers_var.get()
+    gpuid = gpuid_entry.get()
+    gradient_checkpointing = gradient_checkpointing_var.get()
+    grad_accum = grad_accum_entry.get()
+    logdir = logdir_entry.get()
+    log_step = log_step_entry.get()
+    lowvram = lowvram_var.get()
+    resume_ckpt = resume_ckpt_entry.get()
+    run_name = run_name_entry.get()
+    sample_steps = sample_steps_entry.get()
+    save_ckpt_dir = save_ckpt_dir_entry.get()
+    save_every_n_epochs = save_every_n_epochs_entry.get()
+    save_ckpts_from_n_epochs = save_ckpts_from_n_epochs_entry.get()
+    scale_lr = scale_lr_var.get()
+    seed = seed_entry.get()
+    shuffle_tags = shuffle_tags_var.get()
+    validation_config = validation_config_entry.get()
+    wandb = wandb_var.get()
+    write_schedule = write_schedule_var.get()
+
+
+    # Convert integer values to strings
+    resolution = str(resolution_slider.get())
+    batch_size = str(batch_size_entry.get())
+    gradient_steps = str(gradient_steps_entry.get())
+    max_epochs = str(max_epochs_slider.get())
+    sample_steps = str(sample_steps_entry.get())
+    grad_accum = str(grad_accum_entry.get())
+    log_step = str(log_step_entry.get())
+    save_every_n_epochs = str(save_every_n_epochs_entry.get())
+    save_ckpts_from_n_epochs = str(save_ckpts_from_n_epochs_entry.get())
+    seed = str(seed_entry.get())
+
+    # Construct the command
+    command = [
+        'python',
+        'train.py',
+        '--model_type', model_type,
+        '--Save_optimizer_state', str(save_optimizer_state),
+        '--resume', str(resume),
+        '--Project_Name', project_name,
+        '--Learning_Rate', learning_rate,
+        '--Match_text_to_Unet', str(match_text_to_unet),
+        '--Text_lr', text_lr,
+        '--Schedule', schedule,

```

Comment:
not an arg, I think you meant lr_scheduler

Reply:
Ty for the info, this was the updated one. Gradient steps is for the gradient... I can't remember the arg but to increase virtual batch size batch size 4 x 2 gradient steps = 8 per step. 

I'll get the rest of this cleaned up shortly.  Still havnt gotten around to getting ed2 running on local Again.

Reply:
that's grad_accum not gradient_steps.